bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/bridge/AutomowerBridgeHandler.java

```diff
@@ -0,0 +1,160 @@
+
+import static org.openhab.binding.automower.internal.AutomowerBindingConstants.THING_TYPE_BRIDGE;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.automower.internal.rest.api.automowerconnect.dto.MowerListResult;
+import org.openhab.binding.automower.internal.rest.exceptions.AutomowerCommunicationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AutomowerBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public class AutomowerBridgeHandler extends BaseBridgeHandler {
+    private final Logger logger = LoggerFactory.getLogger(AutomowerBridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private @Nullable ScheduledFuture<?> automowerBridgePollingJob;
+    private long automowerPollingIntervalS = TimeUnit.HOURS.toSeconds(2);
+
+    private @NonNullByDefault({}) AutomowerBridge bridge;
+    private @NonNullByDefault({}) AutomowerBridgeConfiguration bridgeConfiguration;
+
+    private final HttpClient httpClient;
+
+    @NonNullByDefault
+    private static class AutomowerBridgePollingRunnable implements Runnable {
+        private final Logger logger = LoggerFactory.getLogger(AutomowerBridgePollingRunnable.class);
+
+        private final AutomowerBridgeHandler handler;
+        private final AutomowerBridge bridge;
+
+        private AutomowerBridgePollingRunnable(AutomowerBridgeHandler handler, AutomowerBridge bridge) {
+            this.handler = handler;
+            this.bridge = bridge;
+        }
+
+        @Override
+        public void run() {
+            MowerListResult automowers;
+            try {
+                automowers = bridge.getAutomowers();
+                handler.updateStatus(ThingStatus.ONLINE);
+                logger.info("Found {} automowers", automowers.getData().size());
+            } catch (AutomowerCommunicationException e) {
+                handler.updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error-query-mowers-failed");
+                logger.warn("Unable to fetch automowers", e);
```

Comment:
Actually yes. The bridge.getAutomowers() might need to do authentication, refresh the authentication token and queries the actual api. The message alone might not give me enough information to distinguish that... But maybe I could log the stacktrace with level debug. What do you think?

Reply:
From the usability perspective the message should be meaningful by itself, so that the user don't need to read the stacktrace to tackle the problem. Also, it seems a bit unreasonable to log a stacktrace when the credentials are incorrect.

Reply:
makes sense